modules/core/src/main/scala/org/tesselation/domain/cluster/storage/AddressStorage.scala

```diff
@@ -1,12 +1,21 @@
 package org.tesselation.domain.cluster.storage
 
+import org.tesselation.infrastructure.db.schema.StoredAddress
 import org.tesselation.schema.address.Address
 import org.tesselation.schema.balance.Balance
 
+sealed trait DBUpsertAction[A] { val value: A }
+case class Insert[A](value: A) extends DBUpsertAction[A]
+case class Update[A](value: A) extends DBUpsertAction[A]
+
 trait AddressStorage[F[_]] {
   def getBalance(address: Address): F[Balance]
 
+  def getMaybeBalance(address: Address): F[Option[Balance]]
```

Comment:
can we combine it into a single method?
can be there `getOrElse(0L)`? or just throw an error if the get query failed 🤔

Reply:
The `getBalance` is used to return a value - it defaults to 0 if there is no balance in the table for that address. I added `getMaybeBalance` to determine whether I need to insert the value or update it during the batch upsert based on transactions in a block. I could implement `getBalance` in terms of `getMaybeBalance` if that helps?

Reply:
Only one is needed. Either what's already done and second method can be "exists", or just the implementation with option